### ISSUE TITLE
pydoc for python statements?

### ISSUE THREAD

**Issue by pjheslin** [jorgenschaefer/elpy#856]
I'm just trying out elpy and it looks great!  I've just found one odd thing.  elpy-doc works great for everything (imported and user-defined functions, etc), but not for basic python statements.  If I try it on "def" for example, the prompt says "Pydoc for: def" and when I hit return it says "no documentation found".  I don't think it's a problem with my config, since elpy-doc works for other, more complicated contexts.  This is elpy-version 1.11.0, on a Mac with python3 installed from homebrew.  pydoc works ok from the command line.

**Comment by jorgenschaefer** [comment]
Hello and thanks for the question! I'm afraid right now the integrated Pydoc support in Elpy only works on variables. The pydoc executable does quite some magic to figure out what to show documentation on, and Elpy did not replicate all of that.

Might be worthwhile to extend it to support that, though – a good idea. :-)

**Comment by pjheslin** [comment]
Ah, I assumed that elpy just called the pydoc executable. So when it did
not work on the same statements, I thought that something was badly wrong
with my configuration. I suspect that I am not the only person who will
(mis)interpret elpy's pydoc prompt in this way.

Thanks again for elpy: it's great!

Peter

**Comment by jorgenschaefer** [comment]
Yeah … Elpy adds some logic – e.g. it supports tab completion there! – but it really should mimic the full interface. :-)

Thanks for the kind words, and enjoy Elpy!
